### PR TITLE
feat: support for tier and tag override configurations

### PR DESCRIPTION
**Pull request by laxmanchekka**
## Description
Added support for configuring tag override configuration and tier configs.

### Testing
Added unit tests. 

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
None

### CODE REVIEW

File: view-generator-framework/src/main/java/org/hypertrace/core/viewgenerator/service/MultiViewGeneratorLauncher.java [hunk 1]
```diff
@@ -104,3 +106,8 @@ private Config getSubJobConfig(ConfigClient client, String jobName) {
         ConfigUtils.getEnvironmentProperty("container.name"));
   }
 }
+
+// common - common gradle module
```
Comment (tim-mwangi) [inline view-generator-framework/src/main/java/org/hypertrace/core/viewgenerator/service/MultiViewGeneratorLauncher.java:110]: remove these or are they important?

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotTableSpec.java [hunk 2]
```diff
@@ -47,6 +47,12 @@
   private String brokerTenant;
   private String serverTenant;
 
+  // Task configs
+  @Optional private Config tagOverrideConfigs;
+
+  // Task configs
```
Comment (laxmanchekka) [inline view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotTableSpec.java:53]:
```suggestion
  // Tier configs
```

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotTableSpec.java [hunk 3]
```diff
@@ -47,6 +47,12 @@
   private String brokerTenant;
   private String serverTenant;
 
+  // Task configs
```
Comment (laxmanchekka) [inline view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotTableSpec.java:50]:
```suggestion
  // Tag override configs
```

File: view-generator-framework/src/main/java/org/hypertrace/core/viewgenerator/service/MultiViewGeneratorLauncher.java (hunk 1 above)
Comment (laxmanchekka) [reply view-generator-framework/src/main/java/org/hypertrace/core/viewgenerator/service/MultiViewGeneratorLauncher.java]: Done

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java [hunk 4]
```diff
@@ -483,6 +529,14 @@ private static String getControllerAddressForTableUpdate(
         "http://%s:%s/%s/%s", controllerHost, controllerPort, PINOT_REST_URI_TABLES, tableName);
   }
 
+  private static String getOptionalString(Config config, String key, String defaultValue) {
+    if (config.hasPath(key)) {
+      return config.getString(key);
+    } else {
```
Comment (rish691) [inline view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:535]: nit: else not needed

**Review by rish691 — DISMISSED** [review]

**Review by rish691 — APPROVED** [review]

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java (hunk 4 above)
Comment (laxmanchekka) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java]: Fixed

**Pull request merged**